node/src/components/deploy_acceptor.rs

```diff
@@ -5,41 +5,126 @@ mod tests;
+    InvalidDeployParameters {
+        prestate_hash: Digest,
+        failure: DeployParameterFailure,
+    },
+}
+
+/// A representation of the way in which a deploy failed validation checks.
+#[derive(Clone, DataSize, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Error, Serialize)]
+pub enum DeployParameterFailure {
+    /// Account does not exist
+    #[error("Account does not exist")]
+    NonexistentAccount { account_hash: AccountHash },
+    /// Nonexistent contract at hash
+    #[error("Contract at {contract_hash} does not exist")]
+    NonexistentContractAtHash { contract_hash: ContractHash },
+    #[error("Contract named {name} does not exist in Account's NamedKeys")]
+    NonexistentContractAtName { name: String },
+    /// Nonexistent contract entrypoint
+    #[error("Contract does not have {entry_point}")]
+    NonexistentContractEntryPoint { entry_point: String },
+    /// Contract Package does not exist
+    #[error("Contract Package at {contract_package_hash} does not exist")]
+    NonexistentContractPackageAtHash {
+        contract_package_hash: ContractPackageHash,
+    },
+    #[error("ContractPackage named {name} does not exist in Account's NamedKeys")]
+    NonexistentContractPackageAtName { name: String },
+    /// Invalid contract at given version.
+    #[error("Invalid contract at version: {contract_version}")]
+    InvalidContractAtVersion { contract_version: ContractVersion },
+    /// Invalid associated keys
+    #[error("Account authorization invalid")]
+    InvalidAssociatedKeys,
+    /// Insufficient deploy signature weight
+    #[error("Insufficient deploy signature weight")]
+    InsufficientDeploySignatureWeight,
     /// A deploy was sent from account with insufficient balance.
-    #[error("insufficient balance")]
+    #[error("Insufficient balance")]
     InsufficientBalance,
+    /// A deploy was sent from account with an unknown balance.
+    #[error("Unable to determine balance")]
+    UnknownBalance,
+    /// Transfer is not valid for payment code.
+    #[error("Transfer is not valid for payment code")]
+    InvalidPaymentVariant,
+    /// Missing payment amount.
+    #[error("missing payment argument amount")]
+    MissingPaymentAmount,
+    /// Failed to parse payment amount.
+    #[error("failed to parse payment amount as U512")]
+    FailedToParsePaymentAmount,
+    /// Failed to parse transfer amount.
+    #[error("failed to parse transfer amount as U512")]
+    FailedToParseTransferAmount,
+    /// Missing transfer amount.
+    #[error("missing transfer amount")]
+    MissingTransferAmount,
+    /// Missing transfer target argument.
+    #[error("missing transfer target argument")]
+    MissingTransferTargetArgument,
+    /// Missing transfer source argument.
+    #[error("missing transfer source argument")]
+    MissingTransferSourceArgument,
+    /// Module bytes for session code cannot be empty.
+    #[error("module bytes for session code cannot be empty")]
+    MissingModuleBytes,
```

Comment:
From experience in working with Kibana - putting the details into the error message is making debugging more difficult as the account hash/prestate/public key/etc. are now part of the generic `"message"` field and not a separate keyword in the log that can be indexed by ElasticSearch.

Reply:
@goral09 would it be possible to do *both*? That way the error message still contains plenty information for the end user, via the responder. I don't believe it has to be one or the other, please correct me if I am wrong here

Reply:
I guess it will be possible to have both but will require some dilligence on the logging side - developer will have to remember about this and construct a correct log format. 

I think you're right - we shouldn't be changing our code b/c _a_ tool we use for logs works this way.